src/types/eval_const.ts

```diff
@@ -364,51 +375,51 @@ function evalBinary(expr: BinaryOperation): Value {
  * TODO: The order of some operations changed in some version.
  * So perhaps to be fully precise here we will need a compiler version too?
  */
-export function evalConstantExpr(expr: Expression): Value {
-    if (!isConstant(expr)) {
-        throw new NonConstantExpressionError(expr);
+export function evalConstantExpr(node: Expression): Value {
+    if (!isConstant(node)) {
+        throw new NonConstantExpressionError(node);
     }
 
-    if (expr instanceof Literal) {
-        return evalLiteral(expr);
+    if (node instanceof Literal) {
+        return evalLiteral(node);
     }
 
-    if (expr instanceof UnaryOperation) {
-        return evalUnary(expr);
+    if (node instanceof UnaryOperation) {
+        return evalUnary(node);
     }
 
-    if (expr instanceof BinaryOperation) {
-        return evalBinary(expr);
+    if (node instanceof BinaryOperation) {
+        return evalBinary(node);
     }
 
-    if (expr instanceof TupleExpression) {
-        return evalConstantExpr(expr.vOriginalComponents[0] as Expression);
+    if (node instanceof TupleExpression) {
+        return evalConstantExpr(node.vOriginalComponents[0] as Expression);
     }
 
-    if (expr instanceof Conditional) {
-        return evalConstantExpr(expr.vCondition)
-            ? evalConstantExpr(expr.vTrueExpression)
-            : evalConstantExpr(expr.vFalseExpression);
+    if (node instanceof Conditional) {
+        return evalConstantExpr(node.vCondition)
+            ? evalConstantExpr(node.vTrueExpression)
+            : evalConstantExpr(node.vFalseExpression);
     }
 
-    if (expr instanceof Identifier) {
-        const decl = expr.vReferencedDeclaration;
+    if (node instanceof Identifier) {
+        const decl = node.vReferencedDeclaration;
 
         if (decl instanceof VariableDeclaration) {
             return evalConstantExpr(decl.vValue as Expression);
         }
     }
 
-    if (expr instanceof FunctionCall) {
+    if (node instanceof FunctionCall) {
```

Comment:
Lets be precise here and check also that `node.kind === FunctionCallKind.typeConversion`

Reply:
The call to `isConstant` does that check.